tools/gulptasks/lint-ts.js

```diff
@@ -30,17 +30,24 @@ const SOURCE_GLOB = './**/*.ts';
 function lintTS(argv) {
     const logLib = require('../libs/log');
     const processLib = require('../libs/process');
+    const utils = require('./utils');
 
-    return new Promise((resolve, reject) => {
-        let product = 'highcharts';
-        let productTSFolder = './ts';
+    let product = argv.product || 'Highcharts';
+    let productTSFolder = './ts';
+
+    if (!utils.validateProduct(product)) {
+        return Promise.resolve();
+    }
 
+    return new Promise((resolve, reject) => {
         if (argv.dashboards) {
-            product = 'dashboards';
+            product = 'Dashboards';
             productTSFolder = './ts/Dashboards';
         } else if (argv.datagrid) {
-            product = 'datagrid';
+            product = 'DataGrid';
             productTSFolder = './ts/DataGrid';
```

Comment:
Is it correct dir?

Reply:
Removed it here: https://github.com/highcharts/highcharts/pull/22655